apps/ios/Shared/Views/ChatList/ChatListView.swift

```diff
@@ -89,15 +126,30 @@ struct UserPickerSheetView: View {
             )
         }
         .onDisappear {
+            let advancedNetworkCanBeSaved = advancedNetworkSettingsCanBeSaved(ss.advancedNetworkSettings)
+            let advancedNetworkSaveText = NSLocalizedString("Save and reconnect", comment: "alert button")
+            
             if serversCanBeSaved(
                 ss.servers.currUserServers,
                 ss.servers.userServers,
                 ss.servers.serverErrors
             ) {
                 showAlert(
-                    title: NSLocalizedString("Save servers?", comment: "alert title"),
-                    buttonTitle: NSLocalizedString("Save", comment: "alert button"),
-                    buttonAction: { saveServers($ss.servers.currUserServers, $ss.servers.userServers) },
+                    title: NSLocalizedString(advancedNetworkCanBeSaved ? "Save servers and network settings?" : "Save servers?", comment: "alert title"),
+                    buttonTitle: advancedNetworkCanBeSaved ? NSLocalizedString("Save", comment: "alert button"): advancedNetworkSaveText,
+                    buttonAction: {
+                        saveServers($ss.servers.currUserServers, $ss.servers.userServers)
```

Comment:
similar to saveNetCfg, saveServers can be refactored to avoid passing bindings and sendable warnings. better in a separate pr after this